From 3513c5358973c738abd89ce8c1d05089fee64765 Mon Sep 17 00:00:00 2001
Message-Id: <3513c5358973c738abd89ce8c1d05089fee64765.1414143271.git.chang-joon.lee@intel.com>
In-Reply-To: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
References: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Wed, 15 Oct 2014 23:28:30 +0530
Subject: [PATCH 10/26] REVERTME [VPG]: drm/i915: Program pixel overlap after
 CHT B0 step

Program pixel overlap only if CHT SoC stepping is >= B0. Older
CHT SoC stepping doesn't support pixel overlap.

REVERTME: This patch can be reverted once everone within Intel has
stopped using CHT SoC steppings older than C0. This patch cannot be
upstreamed as such.

For: GMINL-1532
Change-Id: I507c5cf53bef178b183a8cf196ce3a882d606004
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index f1ec7a7..ea57d89 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -147,7 +147,8 @@ static void intel_dsi_port_enable(struct intel_encoder *encoder)
 					LANE_CONFIGURATION_DUAL_LINK_A;
 
 		if ((intel_dsi->dual_link & MIPI_DUAL_LINK_FRONT_BACK) &&
-							IS_VALLEYVIEW(dev))
+			((IS_CHERRYVIEW(dev_priv->dev) && STEP_FROM(STEP_B0)) ||
+			IS_VALLEYVIEW(dev)))
 			I915_WRITE_BITS(VLV_CHICKEN_3, intel_dsi->pixel_overlap
 					<< PIXEL_OVERLAP_CNT_SHIFT,
 					PIXEL_OVERLAP_CNT_MASK);
-- 
1.7.9.5

